Parse/PFRole.m

```diff
@@ -62,28 +64,17 @@ - (PFRelation *)users {
 ///--------------------------------------
 
 - (void)setObject:(id)object forKey:(NSString *)key {
-    if ([@"name" isEqualToString:key]) {
-        if (self.objectId) {
-            [NSException raise:NSInternalInconsistencyException
-                        format:@"A role's name can only be set before it has been saved."];
-        }
-        if (![object isKindOfClass:[NSString class]]) {
-            [NSException raise:NSInvalidArgumentException
-                        format:@"A role's name must be an NSString."];
-        }
-        if ([object rangeOfString:@"^[0-9a-zA-Z_\\- ]+$" options:NSRegularExpressionSearch].location == NSNotFound) {
-            [NSException raise:NSInvalidArgumentException
-                        format:@"A role's name can only contain alphanumeric characters, _, -, and spaces."];
-        }
+    if ([key isEqualToString:@keypath(PFRole, name)]) {
+        PFConsistencyAssert(self.objectId, @"A role's name can only be set before it has been saved.");
+        PFConsistencyAssert([object isKindOfClass:[NSString class]], @"A role's name must be an NSString.");
+        PFConsistencyAssert([object rangeOfString:@"^[0-9a-zA-Z_\\- ]+$" options:NSRegularExpressionSearch].location != NSNotFound,
+                            @"A role's name can only contain alphanumeric characters, _, -, and spaces.");
     }
     [super setObject:object forKey:key];
 }
 
 - (BFTask *)saveInBackground {
-    if (!self.objectId && !self.name) {
-        [NSException raise:NSInternalInconsistencyException
-                    format:@"New roles must specify a name."];
-    }
+    PFConsistencyAssert(!self.objectId || self.name, @"New roles must specify a name.");
```

Comment:
This seems wrong as well. Shouldn't it be

    self.objectId || self.name

?